Compatibility with OpenSearch 3.6.0

## Description
We need to ensure the compatibility with the next version of OpenSearch v3.6.0
This update is still being discussed, but we need to be aware of potential issues.

For that, we need to:

- [x] Review opensearch's release notes. 
- [x] Identify improvements and potential impact.
- [x] Identify changes on upstream files (listed on https://github.com/wazuh/wazuh-indexer/issues/94)
- [x] Create new development branch.
- [x] Develop a testing environment to verify our components would work under this new build.
- [x] https://github.com/wazuh/wazuh-indexer/issues/1431

## Issues
- https://github.com/wazuh/wazuh-indexer/pull/1454

Builds are currently failing due to the `check_compatibility_versions.sh` script detecting the opensearch version as `3.5.0`:

```shellsession
Run bash build-scripts/check_compatibility_versions.sh
  bash build-scripts/check_compatibility_versions.sh
  shell: /usr/bin/bash -e {0}
  env:
    M2_CACHE_KEY_COMMON: .m2-migrate-main-to-3.6.0-24731410939-common-utils
    M2_CACHE_KEY_ALERTING: .m2-migrate-main-to-3.6.0-24731410939-alerting
    M2_CACHE_KEY_FULL: .m2-migrate-main-to-3.6.0-24731410939-with-security-analytics
    INDEXER_VERSION: 5.0.0
    OPENSEARCH_VERSION: 3.6.0
    PLUGIN_VERSIONS: wazuh-indexer-plugins|5.0.0|3.5.0
  wazuh-indexer-reporting|5.0.0|3.5.0
  wazuh-indexer-security-analytics|5.0.0|3.5.0
  wazuh-indexer-notifications|5.0.0|3.5.0
  wazuh-indexer-common-utils|5.0.0|3.5.0
  wazuh-indexer-alerting|5.0.0|3.5.0
  
OpenSearch version mismatch in wazuh-indexer-plugins: indexer='3.6.0', repo='3.5.0'
OpenSearch version mismatch in wazuh-indexer-reporting: indexer='3.6.0', repo='3.5.0'
OpenSearch version mismatch in wazuh-indexer-security-analytics: indexer='3.6.0', repo='3.5.0'
OpenSearch version mismatch in wazuh-indexer-notifications: indexer='3.6.0', repo='3.5.0'
OpenSearch version mismatch in wazuh-indexer-common-utils: indexer='3.6.0', repo='3.5.0'
OpenSearch version mismatch in wazuh-indexer-alerting: indexer='3.6.0', repo='3.5.0'
Compatibility check failed.
Error: Process completed with exit code 1.
```

--- a/distribution/build.gradle
+++ b/distribution/build.gradle
@@ -373,12 +373,12 @@ configure(subprojects.findAll { ['archives', 'packages'].contains(it.name) }) {
 
     agentFiles = {
       copySpec {
-        from(project(':libs:agent-sm:agent').prepareAgent) {
+        from(configurations.agent) {
           include '**/*.jar'
           exclude '**/*-javadoc.jar'
           exclude '**/*-sources.jar'
           // strip the version since jvm.options is using agent without version
-          rename("opensearch-agent-${project.opensearch}.jar", "opensearch-agent.jar")
+          rename("opensearch-agent-${project.version}.jar", "opensearch-agent.jar")
         }
       }
     }

